manager/controllers/advisory_detail.go

```diff
@@ -53,28 +57,42 @@ func AdvisoryDetailHandler(c *gin.Context) {
 		return
 	}
 
+	resp, err := getAdvisory(advisoryName)
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			LogAndRespNotFound(c, err, "advisory not found")
+		} else {
+			LogAndRespError(c, err, "advisory detail error")
+		}
+		return
+	}
+
+	c.JSON(http.StatusOK, resp)
+}
+
+func getAdvisoryFromDB(advisoryName string) (*AdvisoryDetailResponse, error) {
 	var advisory models.AdvisoryMetadata
 	err := database.Db.Where("name = ?", advisoryName).First(&advisory).Error
+	if err != nil {
+		return nil, err
+	}
+
 	if errors.Is(err, gorm.ErrRecordNotFound) {
-		LogAndRespNotFound(c, err, "advisory not found")
```

Comment:
This is a dead code. On L77 you return if `err != nil`  return so here `err` can be only `nil` and never `gorm.ErrRecordNotFound`.

Reply:
yes, thanks